chatbot/app.py

```
@@ -1,90 +1,124 @@
+
+        probs = torch.softmax(output, dim=1)
+        confidence = probs[0][predicted.item()]
+
+        if confidence.item() > 0.75:
+            for intent in intents["intents"]:
+                if intent["tag"] == tag:
+                    return jsonify({
+                        "response": np.random.choice(intent["responses"])
+                    })
+
+    return jsonify({"response": "Sorry, I didn’t understand that."})
```

Ruff (0.14.14)

[warning] String contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF001)